// antd-mobile/es/checkbox/demo/basic.js
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React from 'react';
import { View, Text } from 'react-native';
import { WhiteSpace, Checkbox, List } from 'antd-mobile';
var AgreeItem = Checkbox.AgreeItem;
var CheckboxItem = Checkbox.CheckboxItem;

var BasicCheckboxExample = function (_React$Component) {
  _inherits(BasicCheckboxExample, _React$Component);

  function BasicCheckboxExample(props, context) {
    _classCallCheck(this, BasicCheckboxExample);

    var _this = _possibleConstructorReturn(this, (BasicCheckboxExample.__proto__ || Object.getPrototypeOf(BasicCheckboxExample)).call(this, props, context));

    _this.state = {
      checkBox1: true,
      agreeItem1: true,
      checkboxItem1: true
    };
    return _this;
  }

  _createClass(BasicCheckboxExample, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      return React.createElement(
        View,
        null,
        React.createElement(
          View,
          { style: { padding: 10 } },
          React.createElement(Checkbox, { checked: this.state.checkBox1, style: { tintColor: '#f00' }, onChange: function onChange(event) {
              _this2.setState({ checkBox1: event.target.checked });
            } }),
          React.createElement(WhiteSpace, null),
          React.createElement(
            Checkbox,
            null,
            'Checkbox'
          ),
          React.createElement(WhiteSpace, null),
          React.createElement(Checkbox, { checked: true, disabled: true }),
          React.createElement(WhiteSpace, null),
          React.createElement(Checkbox, { disabled: true })
        ),
        React.createElement(WhiteSpace, null),
        React.createElement(
          AgreeItem,
          null,
          '\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F\u540C\u610F'
        ),
        React.createElement(WhiteSpace, null),
        React.createElement(
          AgreeItem,
          { checked: this.state.agreeItem1, checkboxStyle: { tintColor: '#f00' }, onChange: function onChange(event) {
              _this2.setState({ agreeItem1: event.target.checked });
            } },
          '\u540C\u610F \u300A\u4FE1\u7528\u652F\u4ED8\u670D\u52A1\u5408\u540C'
        ),
        React.createElement(WhiteSpace, null),
        React.createElement(
          AgreeItem,
          { disabled: true },
          '\u672A\u9009\u4E2D\uFF0C\u4E0D\u53EF\u7F16\u8F91 \u300A\u4FE1\u7528\u652F\u4ED8\u670D\u52A1\u5408\u540C\u300B'
        ),
        React.createElement(WhiteSpace, null),
        React.createElement(
          AgreeItem,
          { checked: true, disabled: true },
          '\u5F3A\u5236\u9009\u4E2D,\u4E0D\u53EF\u7F16\u8F91 \u300A\u4FE1\u7528\u652F\u4ED8\u670D\u52A1\u5408\u540C\u4FE1\u7528\u652F\u4ED8\u670D\u52A1\u5408\u540C\u4FE1\u7528\u652F\u4ED8\u670D\u52A1\u5408\u540C\u300B'
        ),
        React.createElement(
          List,
          { style: { marginTop: 12 } },
          React.createElement(
            Text,
            { style: { marginTop: 12 } },
            '\u8868\u5355\u591A\u9009\u9879\uFF0C\u666E\u901A\u5217\u8868\u4E2D\u591A\u9009\u9879'
          ),
          React.createElement(
            CheckboxItem,
            { checked: this.state.checkboxItem1, onChange: function onChange(event) {
                _this2.setState({ checkboxItem1: event.target.checked });
              } },
            '\u7B7E\u7EA6'
          ),
          React.createElement(
            CheckboxItem,
            null,
            '\u7269\u6599\u94FA\u8BBE'
          ),
          React.createElement(
            CheckboxItem,
            { disabled: true },
            '\u673A\u5177\u7EF4\u62A4\uFF08\u4E0D\u80FD\u9009\uFF09'
          ),
          React.createElement(
            CheckboxItem,
            { disabled: true, checked: true },
            '\u4EA7\u54C1\u95EE\u9898\u89E3\u51B3\uFF08\u5FC5\u9009\uFF09'
          )
        )
      );
    }
  }]);

  return BasicCheckboxExample;
}(React.Component);

export default BasicCheckboxExample;
